src/OpenTelemetry/Trace/Configuration/TracerBuilder.cs

```diff
@@ -50,6 +53,29 @@ public TracerBuilder SetSampler(ISampler sampler)
             return this;
         }
 
+        /// <summary>
+        /// Sets the <see cref="Resource"/> describing the app associated with all traces. Overwrites currently set resource. 
+        /// Please use <see cref="MergeResource(Resource)" /> to add additional labels to the currently set resource.
+        /// </summary>
+        /// <param name="resource">Resource to be associate with all traces.</param>
+        /// <returns>Trace builder for chaining.</returns>
+        public TracerBuilder SetResource(Resource resource)
+        {
+            this.Resource = resource ?? Resource.Empty;
+            return this;
+        }
+
+        /// <summary>
+        /// Merge the provided <see cref="Resource"/> with one currently set by builder.
+        /// </summary>
+        /// <param name="resource">Resource to associate with all traces.</param>
+        /// <returns>Trace builder for chaining.</returns>
+        public TracerBuilder MergeResource(Resource resource)
```

Comment:
why do we need this method? Can user merge all necessary resources and then call SetResource with final one?

Reply:
explained in different comment. I'll put more details in comments for the method